src/hermes_claude_agent_sdk/parity/packets.py

```
@@ -0,0 +1,441 @@
+        attempts = _arr(item["attempts"], "cell.attempts", 3); cap = catalog_caps.get(cap_id); expected_paths = None
+        if catalog is not None:
+            if cap is None or item["scenario_id"] != cap["scenario_id"] or item["source_rows"] != cap["source_rows"] or item["required"] != cap["required"]: _fail("cell does not copy catalog")
+            expected_paths = {name: cap[f"{name}_path"] for name in PATHS}
+        if cap is not None and cap.get("repeat_policy", {"mode": "once"}).get("mode") == "once" and len(attempts) != 1: _fail("once cell must have one attempt")
+        if cap is not None and cap.get("repeat_policy", {"mode": "once"}).get("mode") == "consecutive_3" and not 1 <= len(attempts) <= 3: _fail("repeat cell has invalid attempt count")
+        previous = None; validated_attempts: list[Mapping[str, Any]] = []
+        for ordinal, attempt in enumerate(attempts, 1):
+            if not isinstance(attempt, Mapping) or attempt.get("ordinal") != ordinal: _fail("attempt ordinals are not contiguous")
+            validated = _attempt(attempt, expected_paths, f"cell[{index}].attempt[{ordinal}]", cap.get("expected_trace"))
+            if validated["candidate_sha256"] != candidate["candidate_sha256"]: _fail("attempt candidate differs")
+            if item["session_scope"] == "isolated_cell" and (not validated["fresh_boundary"] or validated["boundary_sha256"] in boundaries): isolation_safe = False; _fail("isolated boundary is not fresh/unique")
```

Ruff (0.16.3)

[error] Multiple statements on one line (colon)

(E701)